cloud-controller-manager/do/loadbalancers.go

```diff
@@ -629,15 +632,7 @@ func (l *loadBalancers) buildLoadBalancerRequest(ctx context.Context, service *v
 
 // buildHealthChecks returns a godo.HealthCheck for service.
 func buildHealthCheck(service *v1.Service) (*godo.HealthCheck, error) {
-	healthCheckPort, err := healthCheckPort(service)
```

Comment:
I cannot think of a proper use case doing that, because as laid out in the description, it should be configured on the k8s service/pod side and not on the LB side.
However, there's a good chance that someone is using these annotation right now which we should treat it as a breaking change if we want to deprecate it.

Reply:
I agree that users are probably using the annotation. The question worth asking is if there's actually any scenario where a customization is making things better. I _currently_ do not recognize one.

We could theoretically keep the old annotations and have them take precedence (at the price of having to keep a good chunk of code around that this PR would otherwise remove).

FWIW, CCM is still at v0.x. While our philosophy has been to not implement changes that will break users knowingly (or with reasonable probability), the current versioning does give us some amount of flexibility to remove configuration annotations if we think there won't be any downsides.

Reply:
I was debating on whether to keep the old flawed behaviour around but I really don't think its necessary. This change should have a positive affect on all customers LBs since we will be health checking the proper components and eliminating many negative edge cases which would have resulted in traffic being dropped.

Reply:
To remain on the safe side, I will add a new annotation `revert-to-old-health-check` that will allow a customer to shift back to the existing health check logic (before this change).